tap/xdp/ipproto.c

```diff
@@ -0,0 +1,79 @@
+// +build ignore
+
+#include <linux/bpf.h>
+#include <linux/in.h>
+#include <linux/udp.h>
+#include <linux/if_ether.h>
+#include <linux/ip.h>
+#include <linux/ipv6.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_endian.h>
+
+
+#define MAX_SOCKS 64
+
+static volatile unsigned const char PROTO;
+static volatile unsigned const char PROTO = IPPROTO_ICMP;
```

Comment:
Why is this `volatile` anyway? `IPPROTO_ICMP` is an enum.

Reply:
Fixed with https://github.com/up9inc/mizu/commit/b4a9be6ad3bdf514ee3bc265fb15b921196730da